pkg/util/test/util.go

```diff
@@ -413,11 +413,45 @@ func UninstallStorageCluster(cluster *corev1.StorageCluster, kubeconfig ...strin
 		if _, err = operatorops.Instance().UpdateStorageCluster(cluster); err != nil {
 			return err
 		}
+
+		defaultCheckReferenceTimeout := 30 * time.Second
+		defaultCheckReferenceInterval := 1 * time.Second
+		if err = checkTelemetrySecretGotOwnerReference(cluster, defaultCheckReferenceTimeout, defaultCheckReferenceInterval); err != nil {
+			return err
+		}
 	}
 
 	return operatorops.Instance().DeleteStorageCluster(cluster.Name, cluster.Namespace)
 }
 
+func checkTelemetrySecretGotOwnerReference(cluster *corev1.StorageCluster, timeout, interval time.Duration) error {
+	t := func() (interface{}, bool, error) {
+		secret, err := coreops.Instance().GetSecret("pure-telemetry-certs", cluster.Namespace)
+		if err != nil {
```

Comment:
as we check when telemetry enabled so the secret must exist. so we retry for all errors. if not found then there's an issue.

Reply:
Added a check for other errors